server/server.go

```diff
@@ -743,13 +745,34 @@ func validateRodeRequirementsForPolicy(mod *ast.Module, regoContent string) []er
 	passBlockExists := false
 	// policy must contains a violations block somewhere in the code
 	violationsBlockExists := false
+	// missing field from result return response
+	returnFieldsExist := false
 
 	for _, r := range mod.Rules {
 		if r.Head.Name == "pass" {
 			passBlockExists = true
 		}
+
 		if r.Head.Name == "violations" && r.Head.Key != nil && r.Head.Key.Value.String() == "result" {
 			violationsBlockExists = true
+			for i, module := range r.Module.Rules {
```

Comment:
Spent probably too long looking at the `ast` module and there might be a way to check this more easily:

```go
// check that there's a violations rule/block
violations := mod.RuleSet("violations") // len(violations) > 0 should mean there's a rule

for _, r := range violations {  
	if r.Head.Key.Value.String() != "result" { // could check for ast.String as well
		continue
	}
	for _, b := range r.Body {
		// find the assignment
		if b.Operator() == nil || b.Operator().String() != "eq" {
			continue
		}
		terms := (b.Terms).([]*ast.Term)
		for i, t := range terms {
			fmt.Println("t", t.String())
			fmt.Printf("%v\n", t.Vars())
			switch a := t.Value.(type) {
			case ast.Object:
                                // look at the previous terms to check that it was assigned to result
				fmt.Println("op", terms[i-2])  // =
				fmt.Println("var", terms[i-1]) // result
				fmt.Println(a.Keys())          // ["pass" "name" "description" "message"]
			default:
				continue
			}
		}
	}
}
```
That may not be quite correct, but I got nerd sniped so I figured I'd post what I found. 

There's also some top-level functions that looked promising, like `ast.WalkRules`, `ast.WalkTerms`.

Reply:
Nice!! This is a lot cleaner. Just like the Cel approach we are using in grafeas-elasticsearch. I'll make the change

Reply:
@alexashley This could use another gander if you have a sec